tests/unit/api/metadata/test_v0.py

```diff
@@ -312,6 +370,85 @@ def test_get_table_metadata_success(self) -> None:
             self.assertEqual(response.status_code, HTTPStatus.OK)
             self.assertCountEqual(data.get('tableData'), self.expected_parsed_metadata)
 
+    @responses.activate
+    def test_get_table_programmatic_metadata_success(self) -> None:
```

Comment:
Instead of needing a new test and a new object, you could add `programmatic_descriptions` to `self.mock_metadata` and `self.expected_parsed_metadata`. Beyond that I think the new tests you'd have to write wouldn't be in this file. 

The new functionality you added can be summarized as:
1. `_marshall_table_full` updates the `programmatic_descriptions` with ` _update_prog_descriptions`
2. ` _update_prog_descriptions` acts updates `progrmmatic_descriptions` according to certain criteria
3. `_sort_prog_descriptions` returns a order value according to certain criteria

Since the idea is to unit test new functionality I would expect to see tests for these new methods. However I notice we are lacking a test file for `metadata_utils`. Overall I'd say we could add `programmatic_descriptions` to the test input and output for `test_get_table_metadata_success`, but beyond that we would want to just make a `test_metadata_utils.py` in `tests/unit/utils` and add a test for those methods you added.

Reply:
Alright, here is what I did, hopefully this is pretty much in line with your recommendations:

- I added programmatic_descriptions to mock_data
- I removed the two separate programmatic_description tests from test_v0
- I added a new test_metadata_utils file and added a single test for marshall_table_full at this time. My thought is that by testing this method, it tests the other private methods.

Reply:
> * My thought is that by testing this method, it tests the other private methods.
 While that is true, it's a good practice for unit tests to isolate logic and explicitly test behaviors. For example if method `A()` calls method `B()` and returns some output `C`, we *could* call method `A()`, check that `C` is returned and consider both `A()` and `B()` tested. 

However with this approach if the logic in `B()` ever changes then the test fails.  Then the question becomes why should a change in  the internal logic for `B()` be able to break a unit test for `A()`? We would address this my mocking `B()` with some fake logic in the unit test. I will tag you in a PR that demonstrates this. 

Thanks for your work on this feature!